test/e2e/dnslookup_test.go

```diff
@@ -0,0 +1,299 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/netip"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rancher-sandbox/rancher-desktop-host-resolver/pkg/helper" //nolint:ignore
+	"github.com/rancher-sandbox/rancher-desktop-host-resolver/test/testdns"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/windows"
+	"golang.zx2c4.com/wireguard/windows/tunnel/winipcfg"
+)
+
+var (
+	// for now we use this maybe this can be an env var from test
+	wslDistroName  = "host-resolver-e2e-test"
+	wslTarballName = "distro-0.21.tar"
+	wslTarballURL  = "https://github.com/rancher-sandbox/rancher-desktop-wsl-distro/releases/download/v0.21/distro-0.21.tar"
+	testSrvAddr    = "127.0.0.1"
+	dnsPort        = "53"
+)
+
+func TestLookupARecords(t *testing.T) { //nolint:funlen
+	tmpDir := t.TempDir()
+
+	t.Logf("Dowloading %v wsl distro tarball", wslTarballName)
+	tarballPath := filepath.Join(tmpDir, wslTarballName)
+
+	err := downloadWSLTarball(tarballPath, wslTarballURL)
+	require.NoErrorf(t, err, "Failed to download wsl distro tarball %v", wslTarballName)
+
+	t.Logf("Creating %v wsl distro", wslDistroName)
+	installCmd := cmdExec(
+		tmpDir,
+		"wsl",
+		"--import",
+		wslDistroName,
+		".",
+		tarballPath)
+	err = installCmd.Run()
+	require.NoErrorf(t, err, "Failed to install distro %v", wslDistroName)
+
+	defer func() {
+		t.Logf("Deleting %v wsl distro", wslDistroName)
+		_, err = cmdRunWithOutput("wsl", "--unregister", wslDistroName)
+		require.NoErrorf(t, err, "Failed to start distro %v", wslDistroName)
+	}()
+
+	// It takes a long time to start a new distro,
+	// 20 sec is a long time but that's actually how long
+	// it takes to start a distro without any flakeyness
+	timeout := time.Second * 20
```

Comment:
```suggestion
	timeout := 20 * time.Second
```
(It reads better as English)